Description: Remove code to find topmost X window in SetTransparent()
 The purpose of this code is unknown, but it may be intended to make WM
 decorations transparent. But it causes a crash when docking an AUI panel.
 And in my testing the decorations are transparent even without this code.
Author: PC
Bug: http://trac.wxwidgets.org/ticket/4841
Applied-Upstream: http://trac.wxwidgets.org/changeset/71275
Last-Update: 2012-06-01

Index: wxWidgets/branches/WX_2_8_BRANCH/src/gtk/toplevel.cpp
===================================================================
--- a/wx/src/gtk/toplevel.cpp (revision 63160)
+++ b/wx/src/gtk/toplevel.cpp (revision 71275)
@@ -1458,24 +1458,4 @@
 #include <X11/Xlib.h>
 
-/* Get the X Window between child and the root window.
-   This should usually be the WM managed XID */
-static Window wxGetTopmostWindowX11(Display *dpy, Window child)
-{
-    Window root, parent;
-    Window* children;
-    unsigned int nchildren;
-
-    XQueryTree(dpy, child, &root, &parent, &children, &nchildren);
-    XFree(children);
-
-    while (parent != root) {
-        child = parent;
-        XQueryTree(dpy, child, &root, &parent, &children, &nchildren);
-        XFree(children);
-    }
-
-    return child;
-}
-
 bool wxTopLevelWindowGTK::SetTransparent(wxByte alpha)
 {
@@ -1484,8 +1464,5 @@
 
     Display* dpy = GDK_WINDOW_XDISPLAY (m_widget->window);
-    // We need to get the X Window that has the root window as the immediate parent
-    // and m_widget->window as a child. This should be the X Window that the WM manages and
-    // from which the opacity property is checked from.
-    Window win = wxGetTopmostWindowX11(dpy, GDK_WINDOW_XID (m_widget->window));
+    Window win = GDK_WINDOW_XID(m_widget->window);
 
     // Using pure Xlib to not have a GTK version check mess due to gtk2.0 not having GdkDisplay
